Library/Homebrew/dev-cmd/unbottled.rb

```diff
@@ -169,7 +169,7 @@ def output_unbottled(formulae, deps_hash, noun, hash, any_named_args)
 
       requirements = f.recursive_requirements
       if @bottle_tag.linux?
-        if requirements.any?(MacOSRequirement)
+        if requirements.any? { |r| r.is_a?(MacOSRequirement) && !r.version_specified? }
```

Comment:
`version_specified?` always returns false when run on Linux I think. `!r.version` should work however.

Reply:
Ah. Missed that since I was testing on macOS. Will push the fix.